g2p/mappings/__init__.py

```diff
@@ -318,18 +318,32 @@ def mapping_to_file(self, output_path: str = GEN_DIR, file_type: str = "json"):
         with open(fn, "w", encoding="utf8", newline="\n") as f:
             self.mapping_to_stream(f, file_type)
 
-    def export_to_dict(self, mapping_type="json"):
-        model_dict = json.loads(
-            self.model_dump_json(exclude_none=True, exclude={"parent_dir": True})
-        )
+    def export_to_dict(self, mapping_type="json", config_only=False):
+        """Export a mapping to a dictionary, optionally including only the
+        configuration (and not the rules or alignments) in the case
+        where we are just writing the config file.
+        """
+        if config_only:
+            model_dict = self.model_dump(
+                mode="json",
+                exclude_none=True,
+                exclude_defaults=True,
```

Comment:
Oh, I don't like this much. It resulted in all the `norm_form: NFD` declarations being removed. While equivalent for the machine, it's less clear for the human reader...

Reply:
Ditto all the `rule_ordering: as-written`

Reply:
Yes, makes sense, I mainly wanted it to remove the `abbreviations` and `alignments` keys, but actually we should just explicitly remove these when `config_only=True` because they should not exist in a configuration file either.  Done in the commit below!